### PR TITLE
Revert "scroll, SetScrollTop, SetScrollLeft in `dom-element-scroll` with test"

### LINKED ISSUE #19268: simple_scroll_to_fragment.html is now a perma-failure
This regressed sometime on November 13, because there were no failures reported prior to that date, and since then there are >60 failures every day. This wasn't noticed because it's marked as intermittent at #14606.

### PR DESCRIPTION
**Pull request by mrobinson**
This allow the WPT test, simple_scroll_to_fragment_regression.html, to
start passing again. Step 10 cannot be supported until a few underlying
issues with our CSSOM implementation are cleared up.

Fixes #19268.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #19268.

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19792)

### CODE REVIEW

**Comment by mrobinson** [comment]
cc @tigercosmos

**Comment by jdm** [comment]
@bors-servo r+
Thanks for dealing with this!

**Comment by emilio** [comment]
Note that fixing #19430 the right way (cc @jonleighton), may also fix this and make the test properly pass.

Maybe it's better to just mark the test as failing?

**Comment by tigercosmos** [comment]
I suspect this change will make some wpt tests failed?

**Comment by mrobinson** [comment]
@tigercosmos Do you mind giving more insight into why it will cause failing tests? The conditionals I removed always return a true case, so scrolling elements from script was non-functional.

**Comment by tigercosmos** [comment]
at least test in #19127 should failed due to this change

**Comment by mrobinson** [comment]
Ah, I see now that the tests that are newly passing rely on overflow being set to "hidden," which is why they did not catch the issue with overflow:scroll. Additionally it seems like the original PR only added these new conditionals to Step 10. I had previously thought it did more. I'm starting to think that the correct thing to do here is to simply revert the original PR and then re-land it with the proper fixes. That will ensure that we maintain test coverage while fixing the underlying issues with the original change. Sorry for the confusion!

**Comment by jdm** [comment]
@bors-servo r-

Warnings are fatal on CI:
```
error: method is never used: `has_scrolling_box`
   --> components\script\dom\element.rs:367:5
    |
367 |     fn has_scrolling_box(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D dead-code` implied by `-D warnings`
error: method is never used: `has_overflow`
   --> components\script\dom\element.rs:374:5
    |
374 |     fn has_overflow(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: method is never used: `overflow_x_is_hidden`
   --> components\script\dom\element.rs:394:5
    |
394 |     fn overflow_x_is_hidden(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: method is never used: `overflow_y_is_hidden`
   --> components\script\dom\element.rs:401:5
    |
401 |     fn overflow_y_is_hidden(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: aborting due to 4 previous errors
error: Could not compile `script`.
```

**Comment by tigercosmos** [comment]
>  The conditionals I removed always return a true case, so scrolling elements from script was non-functional.
@mrobinson I cannot understand why always return `true`?
There are three conditions, only the first one is dummy condition.

**Comment by mrobinson** [comment]
@tigercosmos Sorry, let me explain a bit better. When trying to scroll an element with overflow:scroll via script, `self.has_scrolling_box()` will always return false. The code in question is: 
```
    // https://drafts.csswg.org/cssom-view/#scrolling-box
    fn has_scrolling_box(&self) -> bool {
        // TODO: scrolling mechanism, such as scrollbar (We don't have scrollbar yet)
        //       self.has_scrolling_mechanism()
        self.overflow_x_is_hidden() ||
        self.overflow_y_is_hidden()
    }
```
Thus, elements with overflow:scroll can never be scrolled via script with this set of conditionals.

**Comment by tigercosmos** [comment]
Yes, this is because the spec says so at that time. Is it better fix `has_scrolling_box` with new [SPEC](https://github.com/w3c/csswg-drafts/pull/1540/files)? not only consider the `hidden` and add others in the function.

**Comment by tigercosmos** [comment]
How do you think revert the PR in this PR and I open a new PR later?

**Comment by mrobinson** [comment]
@tigercosmos That works for me. I'll be happy to take a look at your PR when it is ready!

**Comment by tigercosmos** [comment]
I think we should keep test.

File: tests/wpt/mozilla/tests/css/cssom-view/dom-element-scroll.html [hunk 1]
```diff
@@ -1,99 +0,0 @@
-<!doctype html>
```
Comment (jdm) [inline tests/wpt/mozilla/tests/css/cssom-view/dom-element-scroll.html]: There's no need to remove the actual test files in this PR.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by mrobinson** [comment]
Sorry. I figured that's what you meant when you suggested a revert. I didn't interpret it as a partial revert, figuring that maybe you'd want to modify the tests to include testing for overflow:scroll as well. In any case, I don't think I'll be able to get to this today, so feel free to get the test passing on your own in the meantime. I'll close the PR. I should say though that I prefer not to skip this test for more than a brief time, because it is obviously offering some important test coverage.

**Comment by tigercosmos** [comment]
I am confused. So, you want to add `overflow:scroll` test in WPT(Seems we don't have yet)?

**Comment by jdm** [comment]
Keep in mind that simple_scroll_to_fragment.html is marked as an intermittent failure ([and has been since December 2016](https://github.com/servo/servo/issues/14606), so aren't actually protected from regressions on CI anyways.

**Pull request closed without merging**